### PR TITLE
(DOCSP-28889): Kotlin IA Revamp - Manage Users section

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Create new Manage Users section and move existing user pages from App Services section

### Jira

- https://jira.mongodb.org/browse/DOCSP-28889

### Staged Changes

- [new landing page] [Manage Users](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-28889-ia-manage-users/sdk/kotlin/users/)
- [ported] all sub-pages

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/users.txt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+-----------------------
+
+For all authentication providers other than email/password authentication,
+App Services automatically :ref:`registers a user <kotlin-register>`
+the first time a user authenticates.
+When using the email/password authentication provider, you must
+:ref:`manually register a user <kotlin-manage-email-password-users>`.
+If a user authenticates via more than one method,
+you can :ref:`link these user identities <kotlin-link-user-identities>`
+to a single user object.
```
Comment (nlarew) [inline source/sdk/kotlin/users.txt:40]:
```suggestion
If a user authenticates via more than one method,
you can :ref:`link the user identities <kotlin-link-user-identities>`
for each method to a single user account.
```

File: source/sdk/kotlin/users.txt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+- Enable anonymous users to let users access your App Services App without persisting
+  user data.
+
+When you have a logged-in user, you can:
+
+- :ref:`Open a synced realm <kotlin-open-a-synced-realm>` with the user's
+  configuration object
+- :ref:`Run a backend function <kotlin-call-function>` as the logged-in user
+- :ref:`Log the user out <kotlin-logout>`
+- :ref:`Delete a user <kotlin-delete-users>`
```
Comment (nlarew) [inline source/sdk/kotlin/users.txt:69]:
```suggestion
- :ref:`Delete the user's account <kotlin-delete-users>`
```

File: source/sdk/kotlin/users.txt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+.. TODO: Add bullets for changing active user in a multi-user app and removing user 
+.. the device when Work with Multiple Users page is added
```
Comment (nlarew) [inline source/sdk/kotlin/users.txt:72]: Confirming that we intend to merge this TODO

File: source/sdk/kotlin/users.txt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+can bypass the login flow and access the cached user. Use this to open a
+realm or call an Atlas Function upon subsequent app opens.
+
+User Object
+~~~~~~~~~~~
+
+App Services manages sessions with access tokens and refresh tokens.
+Client SDKs supply the logic to manage tokens, and provide them with requests.
+
+.. TODO: uncomment out text and add link to bullet list when user metadata page is added
```
Comment (nlarew) [inline source/sdk/kotlin/users.txt:84]: Confirming that we intend to merge this TODO

File: source/sdk/kotlin/users.txt [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+On successful login, the Kotlin SDK caches credentials on the device. You
+can bypass the login flow and access the cached user. Use this to open a
+realm or call an Atlas Function upon subsequent app opens.
+
+User Object
+~~~~~~~~~~~
+
+App Services manages sessions with access tokens and refresh tokens.
+Client SDKs supply the logic to manage tokens, and provide them with requests.
```
Comment (nlarew) [inline source/sdk/kotlin/users.txt:82]: This feels out of place to me. Tokens are the only thing we mention in this section and they're relevant to User Sessions, not the data in `Realm.User` objects.

Also user objects aren't really a subsection of login/logout so much as their own section w/ custom user data as a subsection.

Could we reframe both of these sections as a "Get User Data" section or similar?

File: source/sdk/kotlin/app-services/overview.txt [hunk 6]
```diff
@@ -18,8 +18,8 @@ You can connect your client apps to the Atlas App Services
 backend using the SDK. Backend functionality includes:
 
 - :doc:`Connect to a Realm App </sdk/kotlin/app-services/connect-to-app-services-backend>`
-- :doc:`Register Users </sdk/kotlin/app-services/register-users>`
-- :doc:`Authenticate Users </sdk/kotlin/app-services/authenticate-users>`
+- :doc:`Register Users </sdk/kotlin/users/register-users>`
+- :doc:`Authenticate Users </sdk/kotlin/users/authenticate-users>`
```
Comment (nlarew) [inline source/sdk/kotlin/app-services/overview.txt:22]: We might convert these to use `:ref:`, up to you if you want to here

File: config/redirects [hunk 7]
```diff
@@ -1038,3 +1039,12 @@ raw: ${prefix}/sdk/react-native/sync-data/check-upload-download-progress ->  ${b
...
 raw: ${prefix}/sdk/react-native/sync-data/pause-or-resume-sync-session ->  ${base}/sdk/react-native/sync-data/manage-sync-session/
+
+# Kotlin IA Revamp (https://jira.mongodb.org/browse/DOCSP-28779)
+
+raw: ${prefix}/sdk/kotlin/app-services/register-users -> ${base}/sdk/kotlin/users/register-users/
+raw: ${prefix}/sdk/kotlin/app-services/authenticate-users -> ${base}/sdk/kotlin/users/authenticate-users/
+raw: ${prefix}/sdk/kotlin/app-services/link-credentials -> ${base}/sdk/kotlin/users/link-credentials/
+raw: ${prefix}/sdk/kotlin/app-services/manage-user-api-keys -> ${base}/sdk/kotlin/users/manage-user-api-keys/
+raw: ${prefix}/sdk/kotlin/app-services/custom-user-data -> ${base}/sdk/kotlin/users/custom-user-data/
+raw: ${prefix}/sdk/kotlin/app-services/delete-users -> ${base}/sdk/kotlin/users/delete-users/
```
Comment (nlarew) [inline config/redirects:1050]:
```suggestion
raw: ${prefix}/sdk/kotlin/app-services/delete-users -> ${base}/sdk/kotlin/users/delete-users/

```

nit: add a newline to the end of the file

**Review by nlarew — COMMENTED** [review]
Back to you with a few suggestions

File: source/sdk/kotlin/users.txt (hunk 5 above)
Comment (cbullinger) [reply source/sdk/kotlin/users.txt]: This page was a shameless steal from Flutter's landing page, so I didn't edit much in terms of content, but good call. I've updated to change the heading levels and tweak this section a bit.

File: source/sdk/kotlin/app-services/overview.txt (hunk 6 above)
Comment (cbullinger) [reply source/sdk/kotlin/app-services/overview.txt]: Done. Thanks for the suggestion

**Review by nlarew — APPROVED** [review]

**Pull request merged**